core/chain_makers.go

```diff
@@ -205,6 +205,12 @@ func GenerateChain(config *params.ChainConfig, parent *types.Block, engine conse
 		b := &BlockGen{i: i, chain: blocks, parent: parent, statedb: statedb, config: config, engine: engine}
 		b.header = makeHeader(chainreader, parent, statedb, b.engine)
 
+		// Set the difficulty for clique block. The chain maker doesn't have access
+		// to a chain, so the difficulty will be left unset (nil). Set it here to the
+		// correct value.
+		if config.TerminalTotalDifficulty == nil && b.header.Difficulty == nil {
+			b.header.Difficulty = big.NewInt(2)
+		}
```

Comment:
This seems an oddball code change. Wasn't the difficulty set on Clique blocks until now?

Reply:
Well yes, the problem is that the chainmaker will set the difficulty for post-merge blocks (which we don't want)

Reply:
* Nope we do want to have difficulty = 0 for post merge blocks (just fixed that)